### PR TITLE
Add Highlighter component

### PR DESCRIPTION
**Pull request by dvoituron**
# Pull Request

## 📖 Description

Add a **Highlighted component** to highlight text in a string based on a search term and using `<mark>` tags.

Highlighter

## 👩‍💻 Reviewer Notes

A sample page is hosted in the Demo project (page /Highlighter).

```xml
<input type="text" @bind-Value="@Highlight" @bind-Value:event="oninput" />

<FluentHighlighter HighlightedText="@Highlight"
                   Delimiters=" ,;"
                   Text="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur unde suscipit, quam beatae rerum inventore consectetur." />

@code 
{
    public string Highlight = "";
}
```

### CODE REVIEW

File: examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Splitter.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+
```
Comment (vnbaaij) [inline examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Splitter.cs:3]: Add above line 4 // Remember to replace the namespace below with your own project's namespace..

File: examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Highlighter.razor.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.AspNetCore.Components;
+using Microsoft.Fast.Components.FluentUI;
+
```
Comment (vnbaaij) [inline examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Highlighter.razor.cs:3]: Add // Remember to replace the namespace below with your own project's namespace..

File: examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Examples/HighlighterDefault.razor [hunk 3]
```diff
@@ -0,0 +1,14 @@
+@namespace FluentUI.Demo.Shared
```
Comment (vnbaaij) [inline examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Examples/HighlighterDefault.razor:1]: Add @* Remember to replace the namespace below with your own project's namespace. *@ at top of file

File: examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Highlighter.razor [hunk 4]
```diff
@@ -0,0 +1,15 @@
+@namespace FluentUI.Demo.Shared
```
Comment (vnbaaij) [inline examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Highlighter.razor:1]: Add @* Remember to replace the namespace below with your own project's namespace. *@ at top of file

**Review by vnbaaij — CHANGES_REQUESTED** [review]

File: examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Splitter.cs (hunk 1 above)
Comment (dvoituron-msft) [reply examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Splitter.cs:3]: Done

File: examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Highlighter.razor.cs (hunk 2 above)
Comment (dvoituron-msft) [reply examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Highlighter.razor.cs:3]: Done

File: examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Examples/HighlighterDefault.razor (hunk 3 above)
Comment (dvoituron-msft) [reply examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Examples/HighlighterDefault.razor:2]: Done

File: examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Highlighter.razor (hunk 4 above)
Comment (dvoituron-msft) [reply examples/FluentUI.Demo.Shared/Pages/Lab/Highlighter/Highlighter.razor:2]: Done

**Review by vnbaaij — APPROVED** [review]

**Pull request merged**